lib/benchee/system.ex

```diff
@@ -32,4 +37,62 @@ defmodule Benchee.System do
     end
   end
 
+  @doc """
+  Returns the number of cores available for the currently running VM.
+  """
+  def num_cores do
+    System.schedulers_online()
+  end
+
+  @doc """
+  Returns an atom representing the platform the VM is running on.
+  """
+  def os do
+    {_, name} = :os.type()
+    os(name)
+  end
+  defp os(:darwin), do: :macOS
+  defp os(:nt), do: :Windows
+  defp os(_), do: :Linux
+
+  @doc """
+  Returns a string with detailed information about the CPU the benchmarks are
+  being performed on.
+  """
+  def cpu_speed, do: cpu_speed(os())
+
+  defp cpu_speed(:Windows), do: "N/A"
+  defp cpu_speed(:macOS), do: system_cmd("sysctl", ["-n", "machdep.cpu.brand_string"])
+  defp cpu_speed(:Linux) do
+    "model name\t:" <> cpu_info = to_string(:os.cmd('cat /proc/cpuinfo | grep "model name"'))
+    String.trim(cpu_info)
+  end
+
+  @doc """
+  Returns an integer with the total number of available memory on the machine
+  running the benchmarks.
+  """
+  def available_memory, do: available_memory(os())
+
+  defp available_memory(:Windows), do: "N/A"
```

Comment:
I think this looks like it should do it for us: http://stackoverflow.com/a/9096023/3877936